x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/StringUtils.java

```diff
@@ -166,6 +168,23 @@ public static String toString(Object value, SqlVersion sqlVersion) {
             return sb.toString();
         }
 
+        // multivalue
+        if (value instanceof Collection) {
+            final StringJoiner sj = new StringJoiner(",", "[", "]");
+            Collection<?> values = (Collection<?>) value;
+            values.forEach(x -> {
+                // quote strings and `\`-escape the `"` character inside them
+                if (x instanceof String) { // TODO: IPs should ideally not be quoted.
+                    sj.add('"' + ((String) x).replace("\"", "\\\"") + '"');
+                } else if (x == null) {
+                    sj.add("NULL");
```

Comment:
Nitpicky: we return uppercase `NULL`, but [here](https://github.com/elastic/elasticsearch/pull/68606/files#diff-c28edb24b62ae39893a3d5f5591e035cfe910e4d042faeec230a34d589da1b74R96) we return lowercase `null` for simple null objects.

Reply:
Sorry, GH only updated the page with your comment after I hit merge.
The link in the comment doesn't seem to resolve correctly, but I assume you refer to the first `if` in the function? 
I followed Posgresql's capitalised formatting for lists. Generally (and debatable) nulls are capitalised in textual formats, so I'd be in favour for updating the listing of simple null objects.

As extra context:
The `NULL` would appear "alone" in CSVs and TSVs -- a simple null object wouldn't be output at all in these formats (they'd be represented by two consecutive commas or tabs).
In CLI, the `NULL` would also appear to the letter in arrays output and the CLI chooses how to represent JSON nulls by itself (which is currently small lettered).
In `txt` formatting output the difference would be visible and "uncontrollable".